pdf.go

```diff
@@ -859,6 +859,54 @@ func (w *pdfPageWriter) EndTextObject() {
 	w.inTextObject = false
 }
 
+// newStringEscaper adds a backlash byte before following bytes: ( ) \
+//
+// According to the PDF spec:
+// When a string is written by enclosing the data in parentheses, bytes whose values are the
+// same as thoseof  the  ASCII  characters  left  parenthesis  (40),  right  parenthesis  (41),
+// and  back-slash  (92)  must  be  preceded  by  a  backslash  character. (...)
+// These  rules  apply  to  each individual  byte  in  a  string  object, whether  the  string
+// is  interpreted  by  the  text-showing operators as single-byte or multiple-byte character codes.
+func newStringEscaper(w io.Writer) escaper {
+	return escaper{
+		w:             w,
+		bytesToEscape: []byte{'\\', '(', ')'},
+		escapeBytes:   []byte{'\\'},
+	}
+}
+
+type escaper struct {
+	w             io.Writer
+	bytesToEscape []byte
+	escapeBytes   []byte
+}
+
+func (r escaper) Write(buf []byte) (n int, err error) {
+	var i, k int
+	for j, b := range buf {
+		if bytes.IndexByte(r.bytesToEscape, b) == -1 {
+			continue
+		}
+		if j != 0 {
+			k, err = r.w.Write(buf[i:j])
+			n += k
```

Comment:
~~You could remove the variable `k` and set `n = j` in the if statement below before returning (and before each `return` after). At the end surely `n = len(p)`.~~ Not if `Write` returns early and you want to return the exact number of bytes written. Was looking at https://golang.org/src/text/tabwriter/tabwriter.go?s=16248:16301#L520 but they use buffer (that never fails).